mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageManager.java

```diff
@@ -634,17 +633,17 @@ private void trimFlags(Flags flags, MailboxSession session) {
 
     protected MessageMetaData appendMessageToStore(final MailboxMessage message, final List<MessageAttachment> messageAttachments, MailboxSession session) throws MailboxException {
         final MessageMapper messageMapper = mapperFactory.getMessageMapper(session);
-        final AttachmentMapper attachmentMapper = mapperFactory.getAttachmentMapper(session);
+
         return mapperFactory.getMessageMapper(session).execute(() -> {
-            ImmutableList.Builder<Attachment> attachments = ImmutableList.builder();
-            for (MessageAttachment attachment : messageAttachments) {
-                attachments.add(attachment.getAttachment());
-            }
-            attachmentMapper.storeAttachmentsForMessage(attachments.build(), message.getMessageId());
+            storeAttachment(message, messageAttachments, session);
             return messageMapper.add(getMailboxEntity(), message);
         });
     }
 
+    protected void storeAttachment(final MailboxMessage message, final List<MessageAttachment> messageAttachments, final MailboxSession session) throws MailboxException {
+
+    }
```

Comment:
so instead of having an Implementation doing nothing (NoopAttachmentMapper), you have a method doing nothing by default? I'm not sure to see the beneficits.
And here you need to duplicate the working method in each effective implementation.

Reply:
The benefit is that **AttachmentMapper** is not advertized anymore in mailbox implementations that do not support it.

The benefits are huge: as a developer I can not do

```
hbaseFactory.getAttachmentMapper(session);
```

BTW maybe my code does nothing for storing attachment, but it do not expose methods for reading them (thus it do not do nothing on reads, as it do not defines read, that's the entire point!)

@mbaechler  may have an opinion on this BTW.

Reply:
If I understand correctly, it's compile-time restriction vs runtime behavior ? You already know my opinion then.

Reply:
even when compile time introduce some strange inheritance? then ok, I will cherry pick this